cmd/csalt/main.go

```diff
@@ -248,6 +281,43 @@ func apiFromArgs(args Args) (*userapi.CacophonyUserAPI, string, error) {
 	return api, saltPrefix, nil
 }
 
+func checkForDuplicates(devices *userapi.DeviceResponse) error {
+	nameMap := make(map[string][]userapi.Device)
+	duplicateNames := make([]string, 0, 1)
+	for _, device := range devices.NameMatches {
+		if _, ok := nameMap[device.DeviceName]; !ok {
+			nameMap[device.DeviceName] = []userapi.Device{device}
+		} else {
+			nameMap[device.DeviceName] = append(nameMap[device.DeviceName], device)
+			duplicateNames = append(duplicateNames, device.DeviceName)
+		}
+	}
+	if len(duplicateNames) > 0 {
+		fmt.Printf("DeviceName Query found %v Duplicate Device please specify group:devicename\n", len(duplicateNames))
+		for _, name := range duplicateNames {
+			fmt.Printf("Device %v matches:\n", name)
+			for _, device := range nameMap[name] {
+				fmt.Printf("%v:%v\n", device.GroupName, device.DeviceName)
+			}
+		}
+		return fmt.Errorf("DeviceName Query found %v Duplicate Device please specify group:devicename\n", len(duplicateNames))
+	}
+	return nil
+}
+
+func showTranslatedDvices(devices *userapi.DeviceResponse) {
+
+	fmt.Println("Translated Name matches:")
+	for _, device := range devices.NameMatches {
+		fmt.Printf("%v:%v saltid: %v\n", device.GroupName, device.DeviceName, device.SaltId)
+	}
+
+	fmt.Println("Translated Devices:")
+	for _, device := range devices.Devices {
+		fmt.Printf("%v:%v saltid: %v\n", device.GroupName, device.DeviceName, device.SaltId)
+	}
```

Comment:
From the user's perspective, do they really need to understand the difference between name matches and device matches? Can these just be shown together?